lib/rules/prefer-destructuring.js

```diff
@@ -15,19 +15,59 @@ module.exports = {
             category: "ECMAScript 6",
             recommended: false
         },
-
         schema: [
             {
-                type: "object",
-                properties: {
-                    array: {
-                        type: "boolean"
+
+                // old support {array: Boolean, object: Boolean}
+                // new support {VariableDeclarator: {}, AssignmentExpression: {}}
+                oneOf: [
+                    {
+                        type: "object",
+                        properties: {
+                            VariableDeclarator: {
+                                type: "object",
+                                properties: {
+                                    array: {
+                                        type: "boolean",
+                                        default: true
```

Comment:
I like it for the clarity, but it has no visible effect from running tests.
I used after checking the examples in https://github.com/json-schema-org/json-schema-org.github.io/search?utf8=%E2%9C%93&q=default%3A&type=
let me know if that’s okay. otherwise, I’m not tied to it and if removing makes more sense, then I can remove.